Plots/Boxplots/NCL_box_3.py

```diff
@@ -0,0 +1,116 @@
+"""
+NCL_box_3.py
+===============
+
+This script illustrates the following concepts:
+   - Drawing box plots
+   - Adding markers to a box plot
+   - Setting the color of individual boxes in a box plot
+   - Setting the width of individual boxes in a box plot
+
+See following URLs to see the reproduced NCL plot & script:
+    - Original NCL script: https://www.ncl.ucar.edu/Applications/Scripts/box_3.ncl
+    - Original NCL plot: https://www.ncl.ucar.edu/Applications/Images/box_3_lg.png
+"""
+
+###############################################################################
+# Import packages:
+import numpy as np
+import matplotlib.pyplot as plt
+
+from geocat.viz import util as gvutil
+
+###############################################################################
+# Generate fake data
+
+np.random.seed(200)
+data = np.random.lognormal(size=(40, 3), mean=1, sigma=.7)
+for a in range(len(data)):
+    data[a] = [x-4 for x in data[a]]
+
+###############################################################################
+# Helper function to set edge color of boxes
+
+
+def setBoxColor(boxplot, number, edge_color):
+
+    # Set edge color of the outside of the boxes and the median lines
+    for element in ['boxes', 'medians']:
+        plt.setp(boxplot[element][number], color=edge_color)
```

Comment:
[This stackoverflow post](https://stackoverflow.com/questions/41997493/python-matplotlib-boxplot-color) talks about coloring box and whisker plots. You may be able to refer to this to figure out how to color the whiskers.

Reply:
I looked at that while I was trying to find a solution, but in all of their examples, every box in each "boxplots" call uses the same colors. When I try to do "for element in ['whiskers', 'caps']" the coloring turns out weird and mismatched, (Like the upper whisker on one box is green and the lower one is blue).

Reply:
Have you considered plotting each box separately and specifying the different colors that way?

Reply:
Figured out how to access each box's actual whiskers and caps, so no need!